src/main.cpp

```diff
@@ -1,105 +1,50 @@
-#include <Arduino.h>
-#include <NativeEthernet.h>
-#include <NativeEthernetUdp.h>
+#include "main.hpp"
 
-#include "pb_encode.h"
-#include "pb_decode.h"
-#include "urc.pb.h"
+int main() {
 
-IntervalTimer encoderMessageTimer;
-elapsedMillis elapsedTime;
+  Context context;
 
-byte mac[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };
-IPAddress ip(192, 168, 8, 255);
-constexpr uint16_t localPort = 8888;
-EthernetUDP Udp;
-
-volatile bool printFlag = false;
-
-void sendEncoderEstimates(DriveEncodersMessage driveEncodersMessage);
-void updateFlag();
-void connectUdp();
-
-void setup()
-{
-  Serial.begin(9600);
-  Serial.println("Hello World!");
-
-  connectUdp();
+  while (true) {
+    updateNetwork(context);
+  }
 
-  encoderMessageTimer.begin([]() { printFlag = true; }, 100000);
+  return 0;
 }
 
-void loop()
-{
-  // create Nanopb messages
-  DriveEncodersMessage driveEncodersMessage = DriveEncodersMessage_init_zero;
+void updateNetwork(Context &context) {
 
-  // read values from drive motor controllers
-  driveEncodersMessage.frontLeftTicks = 12;
-  driveEncodersMessage.frontRightTicks = 15;
-  driveEncodersMessage.middleLeftTicks = 22;
-  driveEncodersMessage.middleRightTicks = 25;
-  driveEncodersMessage.backLeftTicks = 32;
-  driveEncodersMessage.backRightTicks = 35;
+  ethernet_driver::EthernetDriver ethernetDriver = context.getEthernetDriver();
 
-  // sendEncoderEstimates(driveEncodersMessage);
+  // check if ethernet hardware is OK
 
-  // send out encoder message
-  noInterrupts();
-  if (printFlag)
-  {
-    Serial.println(elapsedTime);
-    elapsedTime = 0;
+  // check for incoming messages; if there is one, read it
 
-    Udp.beginPacket(ip, localPort);
-    Udp.write("hello");
-    Udp.endPacket();
+  // send outgoing messages based on TIMER_DURATION
+  // if messages are ready, send them
+  if (ethernet_driver::EthernetDriver::sendTimeHasElapsed()) {
 
-    printFlag = false;
-  }
-  interrupts();
+    // // DEBUG
+    // elapsedMicros driverTimer;
 
-  delay(5);
-}
+    // create Nanopb messages
+    DriveEncodersMessage driveEncodersMessage = DriveEncodersMessage_init_zero;
 
-void connectUdp()
-{
-  // start the Ethernet
-  Ethernet.begin(mac, ip);
+    // read values from drive motor controllers
```

Comment:
Probably clean up the testing in this file before merging

Reply:
We aren't reading any data from a motor controller because we don't have one hooked up. Sending a message with dummy values helps verify code works. Will replace once we can read data from motor controllers.